src/util/submit-addon.js

```diff
@@ -207,10 +208,19 @@ export default class Client {
   }
 
   async doAfterSubmit(addonId, newVersionId, editUrl, patchData) {
+    const promises = [];
     if (patchData && patchData.version) {
       log.info(`Submitting ${Object.keys(patchData.version)} to version`);
-      await this.doFormDataPatch(patchData.version, addonId, newVersionId);
+      promises.push(
+        this.doFormDataPatch(patchData.version, addonId, newVersionId),
+      );
     }
+    if (patchData && patchData.addon) {
+      log.info(`Submitting ${Object.keys(patchData.addon)} to addon`);
+      promises.push(this.doFormDataPatch(patchData.addon, addonId));
+    }
+    await Promise.all(promises);
```

Comment:
The version-specific (source code) data and addon-global (icon) data are independent. A failure in any of these can result in an early termination of the logic. Is that desirable?

Instead of termination as soon as any failure occurs, it may make sense to wrap individual promises in always-fulfilling promises and doing error handling there.

Alternatively, if you'd like to report errors late, use `Promise.allSettled`, and then check whether there are any errors (and print useful information if needed).

Reply:
I think, broadly, proceeding with a warning is probably the way forward - because at that point they can't fix the problem without a whole new version being uploaded (with a version number bump). 

The downside is there isn't a function in `web-ext` to make arbitrary API calls without uploading a versions. So the "fix" would be for them to either make a manual API request (difficult, without the tooling to make the jwt for the Auth); or use devhub (a different UX to command line). And yeah, they might miss the message in the other logging if there isn't a hard error thrown.

Given this is a flaw with the source code upload too, we should probably address that as a separate issue first (blocking V8?)